forge-gui/src/main/java/forge/player/PlayerControllerHuman.java

```diff
@@ -448,11 +448,12 @@ public Integer announceRequirements(final SpellAbility ability, final String ann
         boolean canChooseZero = true;
         Cost cost = ability.getPayCosts();
 
+        if (ability.hasParam("XMaxLimit")) {
```

Comment:
XMaxLimit should only apply when announce = X ?

Reply:
Hmn, I kind of wanted to avoid the extra code for _Tranquil Frillback_
Don't think it's possible for another Announce to get mixed there? 🤔

Reply:
it might be confusing to use XMaxLimit for something not named X

Reply:
Yes, seems this would work. The param name is just a bit unintuitive

Reply:
Alright, guess I'll just switch back to the previous solution to be on the safe side
no big deal anyway